### PR TITLE
log: Avoid breaking single log lines over multiple lines in the log file

### PR DESCRIPTION
**Pull request by maflcko**
Follow up to commit d8b4b3077fd20c90b635eff1dd240bdad9725027

### CODE REVIEW

**Comment by laanwj** [comment]
Code review ACK fa00654e3783d5c2d93ddeb0b06f26fe2ef885d8

File: src/wallet/coinselection.cpp [hunk 1]
```diff
@@ -279,13 +279,13 @@ bool KnapsackSolver(const CAmount& nTargetValue, std::vector<OutputGroup>& group
         }
 
         if (LogAcceptCategory(BCLog::SELECTCOINS)) {
-            LogPrint(BCLog::SELECTCOINS, "SelectCoins() best subset: "); /* Continued */
+            std::string log_message{"SelectCoins() best subset: "};
```
Comment (laanwj) [inline src/wallet/coinselection.cpp:282]: Not introduced in this change, but this function name is wrong. Should be `KnapsackSolver()`? Or simply `__func__`.

Comment (MarcoFalke) [reply src/wallet/coinselection.cpp:282]: I'll let @achow101 decide

Comment (achow101) [reply src/wallet/coinselection.cpp:282]: `__func__`

Comment (MarcoFalke) [reply src/wallet/coinselection.cpp:282]: On a second thought, it is already possible to log the function name with `-logsourcelocations`, so maybe just remove it?

Comment (MarcoFalke) [reply src/wallet/coinselection.cpp]: Resolving discussion for now. Let me know if anything needs to be changed after the latest commit fad8481

File: src/sync.cpp [hunk 2]
```diff
@@ -97,27 +97,29 @@ static void potential_deadlock_detected(const LockPair& mismatch, const LockStac
     LogPrintf("POTENTIAL DEADLOCK DETECTED\n");
     LogPrintf("Previous lock order was:\n");
     for (const LockStackItem& i : s1) {
+        std::string_view prefix{};
```
Comment (practicalswift) [inline src/sync.cpp:100]: Is there any compelling reason to use the "potentially dangerous" `std::string_view` over `std::string` here? (Not claiming that this the use here is dangerous as currently written.)

Lifetime issues are scary and personally I find it _much_ easier to rule out lifetime issues when reviewing code that uses `std::string` compared to code that uses `std::string_view` :)

Somewhat related: [std::string_view encourages use-after-free](https://github.com/isocpp/CppCoreGuidelines/issues/1038)

Comment (MarcoFalke) [reply src/sync.cpp:100]: string_view is no different than span, which is also in our code base. I personally prefer string_view over c-strings (raw pointers).

Luckily, if review doesn't catch use-after-free, we have compilers that check the `lifetimebound` attribute (clang), as well as runtime sanitizers.

Comment (MarcoFalke) [reply src/sync.cpp:100]: Also, this is only used in tests and the program will `std::abort` anyway right after, so I claim it wouldn't matter if a use-after-free made it in (despite all QA).

Comment (MarcoFalke) [reply src/sync.cpp]: Force pushed `std::string` to avoid the discussion :sweat_smile:

**Comment by practicalswift** [comment]
cr ACK 2222c04e1b9960030cb590c789a0d2375add4544

Thanks for removing the use of `std::string_view` ([context](https://github.com/bitcoin/bitcoin/pull/23104#discussion_r718627982)): FWIW that changed my cost-benefit analysis of this PR to transition from `benefit < cost` to `benefit > cost`  :)

**Pull request merged**